src/Microsoft.AspNetCore.SignalR/TypedClientBuilder.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    internal static class TypedClientBuilder<T>
+    {
+        private const string ClientModuleName = "Microsoft.AspNet.SignalR.Hubs.TypedClientBuilder";
+
+        // There is one static instance of _builder per T
+        private static Lazy<Func<IClientProxy, T>> _builder = new Lazy<Func<IClientProxy, T>>(() => GenerateClientBuilder());
+
+        [SuppressMessage("Microsoft.Design", "CA1000:DoNotDeclareStaticMembersOnGenericTypes", Justification = "This is used internally and by tests.")]
+        public static T Build(IClientProxy proxy)
+        {
+            return _builder.Value(proxy);
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1804:RemoveUnusedLocals", MessageId = "forceEvaluation", Justification = "Give me a better way to force the evaluation of Lazy<T>")]
+        public static void Validate()
+        {
+            // The following will throw if T is not a valid type
+            var forceEvaluation = _builder.Value;
```

Comment:
Use `_ = _builder.Value` and remove the `SuppressMessage` attributes. We don't use FxCop in new SignalR ;)